### PR TITLE
Also allow autocompletion within code fragments

### LINKED ISSUE #34: Completion works only at the end of line
If you try calling hackcomplete while in the middle of a line (e.g. for editing parameters in an existing function call), the completion fails. After digging a bit, it seems, that the character under the cursor gets appended to the hh_client call before the magic autocomplete token.

```php
public function morty(): string {
        return 'morty';
} 

public function rick(): void {
        json_encode($this->m);
}
```

Trying to complete `$this->m` (having the cursor pointing at the `)`) will fail, the string passed to hh_client looks like `json_encode($this->)mAUTO332;` If you remove the trailing `);`, the completion will work as expected.

Imho, it should more look like `json_encode($this->mAUTO332);` which correctly completes to `$this->morty`.

I'm not very experienced in vimscript, but I have a solution and will create a pr.

### PR DESCRIPTION
**Pull request by usox**
Fixes #34

### CODE REVIEW

File: autoload/hackcomplete.vim [hunk 1]
```diff
@@ -41,7 +41,7 @@ function! hackcomplete#Complete(findstart, base)
 
   " Insert the base and magic token into the current line.
   let curline = lines[lnum - 1]
-  let lines[lnum - 1] = curline[:cnum - 1] . a:base . s:autotok . curline[cnum :]
+  let lines[lnum - 1] = curline[:cnum - 2] . a:base . s:autotok . curline[cnum - 1] . curline[cnum :]
```
Comment (jwatzman) [inline autoload/hackcomplete.vim]: I also am not a vimscript expert but it looks like this should be equivalent to:

`let lines[lnum - 1] = curline[:cnum - 2] . a:base . s:autotok . curline[cnum - 1 :]`

**Comment by jwatzman** [comment]
Thanks! @mxw is probably the right one to review this.

**Comment by usox** [comment]
@jwatzman Yep, you're right.

**Comment by jwatzman** [comment]
Thanks @usox! Just tested it myself and it seems to work -- I always wondered why autocomplete sometimes didn't work! Thanks for digging in and fixing this. It's simple enough, and I'm confident I've tested it enough, that I'm going to go ahead and merge.

**Comment by mxw** [comment]
Thanks!

**Pull request merged**